lib/rouge/lexers/hql.rb

```diff
@@ -0,0 +1,116 @@
+          LOCATION LOCK LOCKS LOGICAL LONG MAPJOIN MATERIALIZED METADATA MINUS
+          MINUTE MONTH MSCK NOSCAN NO_DROP OFFLINE OPTION OUTPUTDRIVER
+          OUTPUTFORMAT OVERWRITE OWNER PARTITIONED PARTITIONS PLUS PRETTY
+          PRINCIPALS PROTECTION PURGE READ READONLY REBUILD RECORDREADER
+          RECORDWRITER REGEXP RELOAD RENAME REPAIR REPLACE REPLICATION RESTRICT
+          REWRITE RLIKE ROLE ROLES SCHEMA SCHEMAS SECOND SEMI SERDE
+          SERDEPROPERTIES SERVER SETS SHARED SHOW SHOW_DATABASE SKEWED SORT
+          SORTED SSL STATISTICS STORED STREAMTABLE STRING STRUCT TABLES
+          TBLPROPERTIES TEMPORARY TERMINATED TINYINT TOUCH TRANSACTIONS UNARCHIVE
+          UNDO UNIONTYPE UNLOCK UNSET UNSIGNED URI USE UTC UTCTIMESTAMP
+          VALUE_TYPE VIEW WHILE YEAR IF
+
+          ALL ALTER AND ARRAY AS AUTHORIZATION BETWEEN BIGINT BINARY BOOLEAN
+          BOTH BY CASE CAST CHAR COLUMN CONF CREATE CROSS CUBE CURRENT
+          CURRENT_DATE CURRENT_TIMESTAMP CURSOR DATABASE DATE DECIMAL DELETE
+          DESCRIBE DISTINCT DOUBLE DROP ELSE END EXCHANGE EXISTS EXTENDED
+          EXTERNAL FALSE FETCH FLOAT FOLLOWING FOR FROM FULL FUNCTION GRANT
+          GROUP GROUPING HAVING IF IMPORT IN INNER INSERT INT INTERSECT
+          INTERVAL INTO IS JOIN LATERAL LEFT LESS LIKE LOCAL MACRO MAP MORE
+          NONE NOT NULL OF ON OR ORDER OUT OUTER OVER PARTIALSCAN PARTITION
+          PERCENT PRECEDING PRESERVE PROCEDURE RANGE READS REDUCE REVOKE RIGHT
+          ROLLUP ROW ROWS SELECT SET SMALLINT TABLE TABLESAMPLE THEN TIMESTAMP
+          TO TRANSFORM TRIGGER TRUE TRUNCATE UNBOUNDED UNION UNIQUEJOIN UPDATE
+          USER USING UTC_TMESTAMP VALUES VARCHAR WHEN WHERE WINDOW WITH
+
+          AUTOCOMMIT ISOLATION LEVEL OFFSET SNAPSHOT TRANSACTION WORK WRITE
+
+          COMMIT ONLY REGEXP RLIKE ROLLBACK START
+
+          ABORT KEY LAST NORELY NOVALIDATE NULLS RELY VALIDATE
+
+          CACHE CONSTRAINT FOREIGN PRIMARY REFERENCES
+
+          DETAIL DOW EXPRESSION OPERATOR QUARTER SUMMARY VECTORIZATION WEEK YEARS MONTHS WEEKS DAYS HOURS MINUTES SECONDS
+
+          DAYOFWEEK EXTRACT FLOOR INTEGER PRECISION VIEWS
+
+          TIMESTAMPTZ ZONE
+
+          TIME NUMERIC
+        ))
+      end
+
+      def self.keywords_type
+        # source: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Types
+        @keywords_type ||= Set.new(%w(
+            TINYINT SMALLINT INT INTEGER BIGINT FLOAT DOUBLE PRECISION DECIMAL NUMERIC
+            TIMESTAMP DATE INTERVAL
+            STRING VARCHAR CHAR
+            BOOLEAN BINARY
+            ARRAY MAP STRUCT UNIONTYPE
+        ))
+      end
+
+      prepend :root do
+        rule /\$\{/, Name::Variable, :hive_variable
+
+        # The SQL class interprets this as Name::Variable; I'm not 100% sure that's
+        # a bug, but it certainly doesn't agree with what it means in HQL. So I'm
+        # overriding this to mean Str::Single here
+        rule /"/, Str::Single, :double_string
```

Comment:
Seems like a simple choice by the original author of `sql.rb`. If you feel strongly I think it is OK to change it in `sql.rb`. I'm not sure it's worth overriding it here, though. So many of the tokens in Rouge lexers are arbitrary and depend on how the person writing feels or interprets something at that time.

Reply:
From the perspective of hql, it would be worth overriding it because variables can be nested _inside_ strings, so we need to distinguish them. I'll take your suggestion of updating it `sql.rb` and update the pull request?

Reply:
Sure, sounds good.